src/core/ipfs/files.js

```diff
@@ -21,13 +23,18 @@ module.exports = function files (self) {
       const i = new Importer(self._dagS)
       const res = []
 
-      i.on('data', (info) => {
-        res.push(info)
-      })
-
-      i.once('end', () => {
+      // Transform file info tuples to DAGNodes
+      i.pipe(through.obj(function transform (info, enc, next) {
+        console.log('info', info)
+        self._dagS.get(bs58.encode(info.multihash).toString(), (err, node) => {
```

Comment:
What about the ```path:``` field in <a href="https://github.com/ipfs/js-ipfs-unixfs-engine/blob/master/src/importer.js#L53">unixfs-engine importer</a>

Doesn't the cli and using recursive directory adds need this?